exporter/elasticsearchexporter/metadata.yaml

```diff
@@ -9,6 +9,89 @@ status:
   codeowners:
     active: [JaredTan95, carsonip, lahsivjar]
 
+attributes:
+  http_status_code:
+    description: HTTP status code.
+    type: int
+  outcome:
+    description: The operation outcome.
+    type: string
+    enum: [success, failed_client, failed_server, timeout, too_many, failure_store, internal_server_error]
+  greatest_retry:
+    description: The greatest observed retry count in the entire bulk request.
+    type: int
+  failure_store:
+    description: The status of the failure store.
+    type: string
+    enum: [unknown, not_enabled, used, failed]
+
+telemetry:
+  metrics:
+    elasticsearch.bulk_requests.count:
+      prefix: otelcol.
+      stability:
+        level: alpha
+      enabled: true
+      description: Count of the completed bulk requests.
+      unit: "1"
+      sum:
+        value_type: int
+        monotonic: true
+      attributes: [outcome, http_status_code]
+    elasticsearch.docs.received:
+      prefix: otelcol.
+      stability:
+        level: alpha
+      enabled: true
+      description: Count of Elasticsearch docs successfully bufferred to be indexed.
+      unit: "1"
+      sum:
+        value_type: int
+        monotonic: true
+    elasticsearch.docs.indexed:
```

Comment:
will `processed` be better than `indexed`? I assume `indexed` implies successfully indexed, which makes `indexed, outcome=failed_client` a bit strange.